shell/platform/darwin/ios/framework/Headers/FlutterViewController.h

```diff
@@ -165,6 +164,12 @@ FLUTTER_EXPORT
  */
 @property(weak, nonatomic, readonly) FlutterEngine* engine;
 
+/**
+ * The `FlutterBinaryMessenger` associated with this FlutterViewController (used for communicating
+ * with channels).
+ */
+@property(nonatomic, readonly) NSObject<FlutterBinaryMessenger>* binaryMessenger;
```

Comment:
If we're going to break the API, we might as well not leak the implementation detail knowledge of the FlutterEngine (which happens to have a binaryMessenger) into the FlutterViewController.

Reply:
From just looking at FlutterViewController you have no idea you are talking with the engine.  I think we could remove it but it's a slightly more complicated migration.  I tried to be as easy as possible.  If we wanted to go further I'd recommend a separate PR in case someone raises an objection to that change.